src/lightning_app/core/api.py

```diff
@@ -161,86 +167,114 @@ class StateUpdate(BaseModel):
-    update the state."""
-
-    if x_lightning_session_uuid is None:
-        raise Exception("Missing X-Lightning-Session-UUID header")
-    if x_lightning_session_id is None:
-        raise Exception("Missing X-Lightning-Session-ID header")
-
-    body: Dict = await request.json()
-    api_app_delta_queue.put(DeltaRequest(delta=Delta(body["delta"])))
-
-
-@fastapi_service.post("/api/v1/state")
-async def post_state(
-    request: Request,
-    x_lightning_type: Optional[str] = Header(None),
-    x_lightning_session_uuid: Optional[str] = Header(None),
-    x_lightning_session_id: Optional[str] = Header(None),
-) -> None:
-    if x_lightning_session_uuid is None:
-        raise Exception("Missing X-Lightning-Session-UUID header")
-    if x_lightning_session_id is None:
-        raise Exception("Missing X-Lightning-Session-ID header")
-    # This needs to be sent so that it can be set as last state
-    # in app (see sequencing above)
-    # Actually: we need to make sure last_state is actually
-    # the latest state seen by the UI, that is, the last state
-    # ui to the UI from the API, not the last state
-    # obtained by the app.
-    body: Dict = await request.json()
-    x_lightning_session_uuid = TEST_SESSION_UUID
+        layout = deepcopy(state["vars"]["_layout"])
+        for la in layout:
+            if la["content"].startswith("root."):
+                la["content"] = get_component_by_name(la["content"], state)
+        return layout
+
+
+if ENABLE_PULLING_STATE_ENDPOINT:
+
+    @fastapi_service.get("/api/v1/spec", response_class=JSONResponse)
+    async def get_spec(
+        x_lightning_session_uuid: Optional[str] = Header(None),
+        x_lightning_session_id: Optional[str] = Header(None),
+    ) -> List:
+        if x_lightning_session_uuid is None:
+            raise Exception("Missing X-Lightning-Session-UUID header")
+        if x_lightning_session_id is None:
+            raise Exception("Missing X-Lightning-Session-ID header")
+        global app_spec
+        return app_spec or []
+
+
+if ENABLE_PUSHING_STATE_ENDPOINT:
+
+    @fastapi_service.post("/api/v1/delta")
+    async def post_delta(
+        request: Request,
+        x_lightning_type: Optional[str] = Header(None),
+        x_lightning_session_uuid: Optional[str] = Header(None),
+        x_lightning_session_id: Optional[str] = Header(None),
+    ) -> None:
+        """This endpoint is used to make an update to the app state using delta diff, mainly used by streamlit to
+        update the state."""
+
+        if x_lightning_session_uuid is None:
+            raise Exception("Missing X-Lightning-Session-UUID header")
+        if x_lightning_session_id is None:
+            raise Exception("Missing X-Lightning-Session-ID header")
+
+        body: Dict = await request.json()
+        api_app_delta_queue.put(DeltaRequest(delta=Delta(body["delta"])))
+
+
+if ENABLE_PUSHING_STATE_ENDPOINT:
```

Comment:
Might be better to just raise `HTTP 405 Method Not Allowed` instead of removing the endpoint completely.